### ISSUE TITLE
Add ability to filter which WFS/WMS data to display in popup - templates for the Info boxes

### ISSUE THREAD

**Issue by meh9** [TerriaJS/terriajs#496]
Many of the WFS layers have large numbers of key-value pairs that are not relevant to anybody but the custodian of the data, it would be great if we could filter which keys to display in the data popup.

For instance, turning on the GA Island layer and clicking on an island shows the following data:

```
OBJECTID    2970
FEATTYPE    Island
NAME    HINCHINBROOK ISLAND
STATE   QLD
FEATREL 2001-11-28T13:00:00Z
ATTRREL 2001-11-28T13:00:00Z
PLANACC 9999
SOURCE  GEOSCIENCE AUSTRALIA
CREATED 2006-05-08T14:00:00Z
RETIRED 
PID 3491633
SYMBOL  0
SHAPE_Leng  1.32347920603
SHAPE_Area  0.0338646391
```

In this case probably the only keys we are really interested in displaying to an end user might be FEATTYPE, NAME, and STATE. If we do not display all the keys maybe we should have a [...] symbol that could display them all when clicked.

We could control it something like this in the init_nm.json:

```
{
    "name": "Islands",
    ...
    "url": "http://geoserver-nm.nicta.com.au/geotopo_250k/ows",
    "layers": "Framework:islands",
    "displayKeys": "FEATTYPE,NAME,STATE"
}
```

**Comment by kring** [comment]
CC #348 

Even nicer than the list of display keys: we could provide an HTML template to display in the feature popup with something like {FEATTYPE} for feature fields.

**Comment by meh9** [comment]
This is actually just as relevant for WMS services as well.

**Comment by meh9** [comment]
This is becoming a higher and higher priority for AREMI.

@kring having a template for the Info box is definitely the way we would like to take this, it opens up so many possibilities.

**Comment by chloeleichen** [comment]
I can give this a go, do we have a rough wireframe for how it should work? Do we want to allow users to filter data via the UI? Something like the following? 

![info](https://cloud.githubusercontent.com/assets/7508939/10153064/e1de86a2-669b-11e5-89e0-c4728080f10c.png)

**Comment by kring** [comment]
We don't need the user to be able to select the fields (at least not currently!).  I think what AREMI wants (and it's a great TerriaJS feature in general!) is to be able to specify an HTML template in the catalog item definition in the init file.  For example, something like:

```
"infoTemplate": "The name of this station is _{Station Name}_ and its primary fuel source is <em>{Fuel Source - Primary}<em>!"
```

So the curly brace elements in the template get replaced with the value of the corresponding field.  And both HTML and markdown are supported in the template (you'll get this for free just by using our `markdown` Knockout binding).

Using a third-party templating library is fair game, in which case a different placeholder notation for the fields is fine.  But that library needs to be fairly lightweight (we don't want to increase the size of TerriaJS by much for this) and it needs to disallow execution of arbitrary JavaScript code in the placeholders.

**Comment by chloeleichen** [comment]
Does that mean the output into the info box will not necessary display in `<table>`, but can be anything specified in `infoTemplate` ?

**Comment by kring** [comment]
Yep, that's right.

**Comment by kring** [comment]
Fixed by #955
(but see a related issue in #1014)
